File: C++/Plus One.cpp
#include<iostream>
#include<vector>

vector<int> plusOne(vector<int> &digits)
    {
        int len=digits.size();
        int *arry=new int[len];
        for(int i=0;i<len;i++)
            arry[i]=digits[i];
        int pos=digits.size()-1;
        int d=1;
        while(pos>=0)
            {
                int cnt=(arry[pos]+d)/10;
                arry[pos]=(arry[pos]+d)%10;
                d=cnt;
                pos--;
                if(d==0) break;
            }
        vector<int> vec;
        if(d>0) vec.push_back(d);
        for(int i=0;i<len;i++)
            vec.push_back(arry[i]);
        return vec;
    }

File: Python/Plus One.py
#!/usr/bin/env python

class Solution:
    # @param digits, a list of integer digits
    # @return a list of integer digits
    def plusOne(self, digits):
        cnt = 1
        digits.reverse()
        for i in range(len(digits)):
            digits[i] += cnt
            if digits[i] >= 10:
                digits[i] = 0;
            else:
                cnt = 0
                break
        if cnt == 1:
            digits.append(1)
        digits.reverse()
        return digits